metricbeat/helper/prometheus/prometheus_test.go

```diff
@@ -69,6 +69,15 @@ metrics_one_count_total{name="jane",surname="foster"} 1
 metrics_one_count_total{name="john",surname="williams"} 2
 metrics_one_count_total{name="jahn",surname="baldwin",age="30"} 3
 
+`
+
+	promGaugeKeyLabelWithNaNInf = `
+# TYPE metrics_one_count_total gauge
```

Comment:
Maybe add here some case of metric with the same labels than the discarded ones, to check that metrics with the same labels are still being collected.
```suggestion
# TYPE metrics_one_count_errors gauge
metrics_one_count_errors{name="jane",surname="foster"} 0
# TYPE metrics_one_count_total gauge
```